spec/requests/api/articles_index_spec.rb

```diff
@@ -2,14 +2,36 @@
   subject { response }
 
   describe 'when there are some article in the database' do
-    let!(:articles) { create_list(:article, 2) }
-    before do
-      get '/api/articles'
+    let(:tech) { create(:category, name: 'Tech') }
+    let!(:article1) { create(:article, category_id: tech.id, category_name: tech.name) }
+    let!(:article2) { create(:article, category_id: tech.id, category_name: tech.name) }
+    let!(:article3) { create(:article) }
+
+    describe 'searching for all articles' do
+      before do
+        get '/api/articles'
+      end
+      it { is_expected.to have_http_status 200 }
+
+      it 'is expected to return a collection of articles' do
+        expect(response_json['articles'].count).to eq 3
+      end
+
+      it 'is expected to include a category value' do
+        expect(response_json['articles'].last['category_name']).to eq 'MyCategory'
+      end
     end
-    it { is_expected.to have_http_status 200 }
 
-    it 'is expected to return a collection of articles' do
-      expect(response_json['articles'].count).to eq 2
+    describe 'search for articles by categories' do
+      before do
+        get '/api/articles/',
+            params: { category_name: tech.name }
+      end
+      it { is_expected.to have_http_status 200 }
+
+      it 'is expected to return a collection of articles' do
+        expect(response_json['articles'].count).to eq 2
+      end
```

Comment:
What is you have non existent category in params?

Reply:
We have changed it to return all articles if you enter a non existent category